instrumentation/graphql/lib/opentelemetry/instrumentation/graphql/tracers/graphql_tracer.rb

```diff
@@ -86,24 +86,60 @@ def config
-              attributes['graphql.lazy'] = key == 'resolve_type_lazy'
+            when 'execute_field'
+              field_attr_cache = data[:query].context.namespace(:otel_attrs)[:execute_field_attrs] ||= attr_cache do |field|
+                attrs = {}
+                attrs['graphql.field.parent'] = field.owner.graphql_name if field.owner.graphql_name
+                attrs['graphql.field.name'] = field.graphql_name if field.graphql_name
+                attrs['graphql.lazy'] = false
+                attrs.freeze
+              end
+              field_attr_cache[data[:field]]
+            when 'execute_field_lazy'
+              lazy_field_attr_cache = data[:query].context.namespace(:otel_attrs)[:execute_field_lazy_attrs] ||= attr_cache do |field|
+                attrs = {}
+                attrs['graphql.field.parent'] = field.owner.graphql_name if field.owner.graphql_name
+                attrs['graphql.field.name'] = field.graphql_name if field.graphql_name
+                attrs['graphql.lazy'] = true
+                attrs.freeze
+              end
+              lazy_field_attr_cache[data[:field]]
+            when 'authorized', 'resolve_type'
+              type_attrs_cache = data[:context].namespace(:otel_attrs)[:type_attrs] ||= attr_cache do |type|
+                attrs = {}
+                attrs['graphql.type.name'] = type.graphql_name if type.graphql_name
+                attrs['graphql.lazy'] = false
+                attrs.freeze
+              end
+              type_attrs_cache[data[:type]]
+            when 'authorized_lazy', 'resolve_type_lazy'
+              type_lazy_attrs_cache = data[:context].namespace(:otel_attrs)[:type_lazy_attrs] ||= attr_cache do |type|
+                attrs = {}
+                attrs['graphql.type.name'] = type.graphql_name if type.graphql_name
+                attrs['graphql.lazy'] = true
+                attrs.freeze
+              end
+              type_lazy_attrs_cache[data[:type]]
             when 'execute_query'
-              attributes['graphql.operation.name'] = data[:query].selected_operation_name if data[:query].selected_operation_name
-              attributes['graphql.operation.type'] = data[:query].selected_operation.operation_type
-              attributes['graphql.document'] = data[:query].query_string
+              attrs = {}
+              attrs['graphql.document'] = data[:query].query_string if data[:query].query_string
+              # rubocop:disable Style/SafeNavigation - using safe navigation creates more objects, we want to avoid this
+              attrs['graphql.operation.type'] = data[:query].selected_operation.operation_type if data[:query].selected_operation && data[:query].selected_operation.operation_type
+              # rubocop:enable Style/SafeNavigation
+              attrs['graphql.operation.name'] = data[:query].selected_operation_name || 'anonymous'
+              attrs.freeze
+            else
+              {}
```

Comment:
I think this should be a frozen hash and a constant. That will mean only allocating one hash for all "misses"